pkg/interceptors/github/github.go

```diff
@@ -96,7 +118,209 @@ func (w *Interceptor) Process(ctx context.Context, r *triggersv1.InterceptorRequ
 		}
 	}
 
+	if p.AddChangedFiles.Enabled {
+		shouldAddChangedFiles := false
+		for _, allowedEvent := range acceptedEventTypes {
+			if actualEvent == allowedEvent {
+				shouldAddChangedFiles = true
+				break
+			}
+		}
+		if !shouldAddChangedFiles {
+			return &triggersv1.InterceptorResponse{
+				Continue: true,
+			}
+		}
+
+		if r.Context == nil {
+			return interceptors.Failf(codes.InvalidArgument, "no request context passed")
+		}
+
+		secretToken, err := w.getGithubTokenSecret(ctx, r, p)
+		if err != nil {
+			return interceptors.Failf(codes.FailedPrecondition, "error getting secret: %v", err)
+		}
+
+		payload, err := parseBodyForChangedFiles(r.Body, actualEvent)
+		if err != nil {
+			return interceptors.Failf(codes.FailedPrecondition, "error parsing body: %v", err)
+		}
+
+		var changedFiles string
+		if actualEvent == "pull_request" {
+			changedFiles, err = getChangedFilesFromPr(ctx, payload, headers.Get("X-Github-Enterprise-Host"), secretToken)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error getting changed files: %v", err)
+			}
+		} else {
+			changedFiles = payload.ChangedFiles
+		}
+
+		return &triggersv1.InterceptorResponse{
+			Extensions: map[string]interface{}{
+				changedFilesExtensionsKey: changedFiles,
+			},
+			Continue: true,
+		}
+
+	}
+
 	return &triggersv1.InterceptorResponse{
 		Continue: true,
 	}
 }
+
+func (w *Interceptor) getGithubTokenSecret(ctx context.Context, r *triggersv1.InterceptorRequest, p triggersv1.GitHubInterceptor) (string, error) {
+	if p.AddChangedFiles.PersonalAccessToken == nil {
+		return "", nil
```

Comment:
If no PersonalAccessToken passed then we should return error :thinking: asking because if no PersonalAccessToken then further we cannot do anything right 
Or am i missing something here?

Reply:
If the repository does not require authentication, the PersonalAccessToken is not required to make the github API call. This can be seen in the `github-add-changed-files-push-cel` example which makes the github API call without the PAT.